File: src/codeforces/contest1795/ProblemA.java
package codeforces.contest1795;

import java.util.Scanner;

public class ProblemA {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt();
        scanner.nextLine();
        for (int i = 0; i < t; i++) {
            scanner.nextLine();
            String s1 = scanner.nextLine();
            String s2 = scanner.nextLine();
            boolean ans = solve(s1, s2);
            System.out.println(ans ? "YES" : "NO");
        }
    }

    private static boolean solve(String s1, String s2) {
        int cnt = 0;
        int n1 = s1.length(), n2 = s2.length();
        for (int i = 1; i < n1; i++) if (s1.charAt(i - 1) == s1.charAt(i)) cnt++;
        for (int i = 1; i < n2; i++) if (s2.charAt(i - 1) == s2.charAt(i)) cnt++;
        if (s1.charAt(n1 - 1) == s2.charAt(n2 - 1)) cnt++;
        return cnt < 2;
    }
}

File: src/codeforces/contest251/ProblemA.kt
@file:Suppress("DuplicatedCode")

package codeforces.contest251

import java.io.BufferedReader
import java.io.InputStream
import java.io.InputStreamReader
import java.util.*

fun main(args: Array<String>) {
    val n = nextInt()
    val d = nextInt()
    println(solve(nextInts(n), d))
}

fun solve(arr: IntArray, d: Int): Long {
    var ans = 0L
    var l = 0
    for ((r, num) in arr.withIndex()) {
        while (num - arr[l] > d) l++
        ans += (r - l) * (r - l - 1L) / 2
    }
    return ans
}


private class MyReader(inputStream: InputStream) {
    private val reader = BufferedReader(InputStreamReader(inputStream))
    private var tokenizer: StringTokenizer? = null

    fun next(): String {
        while (true) {
            tokenizer.let {
                if (it == null || !it.hasMoreTokens()) {
                    tokenizer = StringTokenizer(reader.readLine())
                } else {
                    return it.nextToken()
                }
            }
        }
    }
}

private val reader = MyReader(System.`in`)
private fun next() = reader.next()
private fun nextInt() = next().toInt()
private fun nextLong() = next().toLong()
private fun nextDouble() = next().toDouble()
private fun nextStrings(n: Int) = Array(n) { next() }
private fun nextInts(n: Int) = IntArray(n) { nextInt() }
private fun nextLongs(n: Int) = LongArray(n) { nextLong() }
private fun nextDoubles(n: Int) = DoubleArray(n) { nextDouble() }